src/api/utils/index.js

```diff
@@ -19,6 +19,48 @@ const windowLoaded = new Promise((resolve) => {
   })
 })
 
+export const web3CompatibleNetwork = async () => {
+  await windowLoaded
+  // blocks access via load
+  if (typeof window === 'undefined' || !window.web3) return (console.error('No Provider detected. Returning UNKNOWN network.'), 'UNKNOWN')
+
+  let { web3 } = window
+  let netID
+
+  // irregular APIs - Opera, new MM, some other providers
+  if (web3.currentProvider && !web3.version) {
+    const Web3 = require('web3')
+    console.warn('Non-Metamask or Gnosis Safe Provider injected web3 API detected')
+
+    window.web3 = web3 = new Web3(web3.currentProvider)
+  }
+
+  // 1.X.X API
+  if (typeof web3.version === 'string') {
+    netID = await new Promise((accept, reject) => {
+      web3.eth.net.getId((err, res) => {
+        if (err) {
```

Comment:
Doesn't `web3.eth.net.getId()` return a Promise by itself if you don't pass in a callback?

Reply:
it does now, i believe it didnt even in 1.0 when i made this, ill change it